### PR TITLE
fix(package-manager): handle npm-alias dependency specs

### LINKED ISSUE #312: panic on npm-alias dependency specs in install (Package::pinned_version)
## What happens

`pacquet install` panics when a workspace depends on a package using an npm-alias spec:

```
Error:   × Main thread panicked.
  ├─▶ at crates/registry/src/package.rs:55:63
  ╰─▶ called `Result::unwrap()` on an `Err` value: SemverError { input:
      "npm:strip-ansi@^6.0.1", span: SourceSpan { offset: SourceOffset(0),
      length: 0 }, kind: NoValidRanges }
```

## Where

[`crates/registry/src/package.rs:55`](https://github.com/pnpm/pacquet/blob/3ea69a6/crates/registry/src/package.rs#L54-L56):

```rust
pub fn pinned_version(&self, version_range: &str) -> Option<&PackageVersion> {
    let range: node_semver::Range = version_range.parse().unwrap(); // TODO: this step should have happened in PackageManifest
```

The whole alias spec (`npm:strip-ansi@^6.0.1`) is fed straight into `node_semver::Range::parse`, which understandably rejects it. The existing TODO already acknowledges that this parse should have happened earlier — the install path simply isn't aware that an npm-alias is a thing.

## Reproduce

Any `package.json` dep using the `npm:<name>@<range>` form, e.g.

```json
{
  "dependencies": {
    "ansi-strip": "npm:strip-ansi@^6.0.1"
  }
}
```

Then `pacquet install`.

## What's expected

pnpm resolves npm-aliases by treating `<name>` from `npm:<name>@<range>` as the registry package name and `<range>` as the version range, while keeping the local key (`ansi-strip` above) as the install path. Lockfile snapshots already learned about npm-aliases in [`feat(lockfile): support npm-alias dependencies in snapshots`](https://github.com/pnpm/pacquet/commit/e5d0e85), but the registry/install side doesn't yet.

Upstream reference for the alias semantics: [pnpm's parsePref / npm-alias handling in `@pnpm/parse-wanted-dependency` and `@pnpm/npm-resolver`](https://github.com/pnpm/pnpm/tree/main/resolving/npm-resolver).

## Suggested direction

1. Detect the `npm:` prefix when reading `package.json` dependencies (manifest layer).
2. Decompose into `(alias_name -> (real_name, version_range))`; persist that distinction through resolution and lockfile.
3. Pass only the parsed version range to `Package::pinned_version`, fetching the registry doc for the real package name.

## Workaround

Until this lands, swap any `"foo": "npm:bar@^X.Y.Z"` deps for the direct `"bar": "^X.Y.Z"` form, accepting the loss of the local rename.

### PR DESCRIPTION
**Pull request by zkochan**
## Summary

- `pacquet install` panicked on any `package.json` dep written as `"foo": "npm:bar@<range>"` because the whole alias spec was fed to `node_semver::Range::parse`. Decompose the spec at the install layer so the registry sees the real package name and the resolver only sees the range — mirroring pnpm's [`parseBareSpecifier`](https://github.com/pnpm/pnpm/blob/1819226b51/resolving/npm-resolver/src/parseBareSpecifier.ts).
- The manifest key stays as the directory name under `node_modules/` (the alias the parent's source code imports). The virtual store keeps using the registry-returned name, so an aliased and a plain entry pointing at the same target share one virtual store directory.
- Unversioned `npm:foo` / `npm:@scope/foo` default to the `latest` tag, matching pnpm's `index < 1` branch.

Closes #312.

## Test plan

- [x] New unit tests in `pacquet-package-manifest` cover plain specs, scoped targets, unversioned `npm:` aliases (default to `latest`), and the `lastIndexOf('@')` edge case.
- [x] New end-to-end test `npm_alias_dependency_installs_under_alias_key` in `pacquet-package-manager` installs `"hello-world-alias": "npm:@pnpm.e2e/hello-world-js-bin@1.0.0"` against the registry mock and asserts the symlink lives under the alias key, the real package name is *not* exposed alongside, and the virtual store uses the real name.
- [x] `just ready` (typos, fmt, check, test, lint) — green.

### CODE REVIEW

**Review by anonrig — APPROVED** [review]

File: crates/package-manager/src/install_package_from_registry.rs [hunk 1]
```diff
@@ -60,9 +73,10 @@ impl<'a> InstallPackageFromRegistry<'a> {
...
-            let package = Package::fetch_from_registry(name, http_client, &config.registry)
-                .await
-                .map_err(InstallPackageFromRegistryError::FetchFromRegistry)?;
+            let package =
+                Package::fetch_from_registry(registry_name, http_client, &config.registry)
+                    .await
+                    .map_err(InstallPackageFromRegistryError::FetchFromRegistry)?;
             let package_version = package.pinned_version(version_range).unwrap(); // TODO: propagate error for when no version satisfies range
             self.install_package_version(package_version).await?;
             package_version.clone()
```
[the remark this answers is not in the document]
Comment (zkochan) [reply crates/package-manager/src/install_package_from_registry.rs:85]: @copilot apply changes based on [this feedback](https://github.com/pnpm/pacquet/pull/323#discussion_r3154426232)
